Makefile

```diff
@@ -0,0 +1,35 @@
+SHELL := bash
+
+PYTHON := $(shell command -v python || command -v python3)
+$(if $(PYTHON),,$(error No python or python3 binary found))
+
+export PYTHONPATH := $(PWD)
+
+PYTHON_TESTS := \
+    test_interpreter \
+    test_vm_compiler \
+    test_barevm \
+
+BAREVM_TESTS := \
+    test_barevm \
+    barevm_unittest \
+
+
+default:
+
+test: $(PYTHON_TESTS) barevm_unittest
+
+$(BAREVM_TESTS):: barevm/barevm
+
+$(PYTHON_TESTS)::
+	$(PYTHON) tests_full/$@.py
+
+barevm_unittest::
```

Comment:
Not sure what this is for, now.

You already have the rule for building barevm. This also builds barevm_unittest. So the test rule should only run the test? (and depend on the build rule to ensure it's there?)

Thinking a bit more about this, most people probably won't want to build barevm at all; the default should be just to run the python tests, with a separate rule to build&test barevm.

Reply:
The `::` rules allow more that one rule for a target name.

Both `BAREVM_TESTS` require barevm/barevm to be built,
but `test_barevm` is a PYTHON_TEST and `barevm_unittest` is it's own thing.

I'll push another commit for your thinking.

Reply:
Fixed in 94d18df